test/test.js

```diff
@@ -9,43 +9,63 @@ const casesDir = path.resolve(__dirname, "cases");
 const implsDir = path.resolve(__dirname, "implementations");
 const outputDir = path.resolve(__dirname, "output");
 
-beforeAll(() => {
-  const transformer = new Transformer({
-    processCEReactions(code) {
-      const ceReactions = this.addImport("../CEReactions");
-
-      return `
-        ${ceReactions}.preSteps(globalObject);
-        try {
-          ${code}
-        } finally {
-          ${ceReactions}.postSteps(globalObject);
-        }
-      `;
-    },
-    processHTMLConstructor() {
-      const htmlConstructor = this.addImport("../HTMLConstructor", "HTMLConstructor");
-
-      return `
-        return ${htmlConstructor}(globalObject, interfaceName);
-      `;
-    }
+const idlFiles = fs.readdirSync(casesDir);
+
+describe("without processors", () => {
+  beforeAll(() => {
+    const transformer = new Transformer();
+    transformer.addSource(casesDir, implsDir);
+
+    return transformer.generate(outputDir);
   });
-  transformer.addSource(casesDir, implsDir);
 
-  return transformer.generate(outputDir);
+  for (const idlFile of idlFiles) {
+    test(idlFile, () => {
+      const outputFile = path.resolve(outputDir, path.basename(idlFile, ".webidl") + ".js");
+      const output = fs.readFileSync(outputFile, { encoding: "utf-8" });
+
+      expect(output).toMatchSnapshot();
+    });
+  }
 });
 
-const idlFiles = fs.readdirSync(casesDir);
+describe("with processors", () => {
```

Comment:
This change doubles the number of tests to validate the transform output with and without processors, which I am not a fan of. It would be preferable to only run the processor tests against the `CEReaction.webidl` and `HTMLConstructor.webidl`, but, it is currently impossible with the available webidl2js API.

Reply:
I have always found the current API a bit strange. I think we should probably go this route for now and do a release, but point to https://github.com/jsdom/webidl2js/issues/160 as a potential way of making it better.

I'll hold off on doing a release for now, in case you think we should tackle #160 first.

Reply:
Agreed, let's move forward with custom elements first. We can always come back to #160 latter.